src/collections/Menu/MenuItem.js

```diff
@@ -8,16 +8,21 @@ import {
 } from '../../lib'
 
 function MenuItem(props) {
-  const { active, children, className, header } = props
+  const { active, children, className, header, href, link, onClick } = props
   const classes = cx(
     className,
     useKeyOnly(active, 'active'),
     useKeyOnly(header, 'header'),
+    useKeyOnly(link, 'link'),
     'item',
   )
+  const handleClick = (e) => {
+    if (onClick) onClick(e)
+  }
   const rest = getUnhandledProps(MenuItem, props)
+  const ItemComponent = href || onClick ? 'a' : 'div'
 
-  return <a {...rest} className={classes}>{children}</a>
+  return <ItemComponent {...rest} className={classes} onClick={handleClick}>{children}</ItemComponent>
```

Comment:
Good catch.  It looks like links are perhaps not quite consistent enough to render the `a` tag across all components.  Though, having the `getElementType()` pattern would still be of value as it would allow us to standardize the `as` prop and render `a` tags for `href`.  Also, as we find any new logic for conditionally rendering element types, it would go here.

You are correct though, the `link` prop should be handled individually by each component.  Here is some more context on how it is used in SUI:


### Steps

```html
<a class="step">

<div class="link step">
```
>http://semantic-ui.com/elements/step.html#link


### Card:

```html
<a class="ui card" href="http://www.dog.com">
<div class="ui link card">
```
>http://semantic-ui.com/views/card.html#link-card

### Label:

```html
<a class="ui label"></a>

<div class="ui label">
  <a class="detail">View Mail</a>
</div>
```
>http://semantic-ui.com/elements/label.html#link


### Image:

```html
<a href="http://google.com" class="ui medium image">
  <img src="/images/wireframe/image-text.png">
</a>
```
>http://semantic-ui.com/elements/image.html#image-link

### List:

```html
<div class="ui link list">
  <div class="active item">Home</div>
  <a class="item">About</a>
  <a class="item">Jobs</a>
  <a class="item">Team</a>
</div>
```
>http://semantic-ui.com/elements/list.html#link